crates/biome_js_analyze/tests/specs/complexity/useOptionalChain/complexLogicalAndCases.ts.snap

```diff
@@ -68,16 +75,80 @@ complexLogicalAndCases.ts:4:1 lint/complexity/useOptionalChain  FIXABLE  ━━
   > 4 │ foo && foo[typeof bar] && foo[typeof bar].baz;
       │ ^^^^^^^^^^^^^^^^^^^^^^
     5 │ 
+    6 │ // https://github.com/biomejs/biome/issues/1925
   
   i Unsafe fix: Change to an optional chain.
   
-    2 2 │   foo && foo[bar as string] && foo[bar as string].baz;
-    3 3 │   foo && foo[1 + 2] && foo[1 + 2].baz;
-    4   │ - foo·&&·foo[typeof·bar]·&&·foo[typeof·bar].baz;
-      4 │ + foo?.[typeof·bar]·&&·foo[typeof·bar].baz;
-    5 5 │   
+     2  2 │   foo && foo[bar as string] && foo[bar as string].baz;
+     3  3 │   foo && foo[1 + 2] && foo[1 + 2].baz;
+     4    │ - foo·&&·foo[typeof·bar]·&&·foo[typeof·bar].baz;
+        4 │ + foo?.[typeof·bar]·&&·foo[typeof·bar].baz;
+     5  5 │   
+     6  6 │   // https://github.com/biomejs/biome/issues/1925
   
 
 ```
 
+```
+complexLogicalAndCases.ts:7:1 lint/complexity/useOptionalChain  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━
 
+  ! Change to an optional chain.
+  
+    6 │ // https://github.com/biomejs/biome/issues/1925
+  > 7 │ foo?.bar && foo.bar?.baz && foo.bar?.qux;
+      │ ^^^^^^^^^^^^^^^^^^^^^^^^
+    8 │ 
+    9 │ foo?.bar.baz?.qux && foo.bar.baz.qux.zoo;
+  
+  i Unsafe fix: Change to an optional chain.
+  
+    7 │ foo?.bar·&&·foo.bar?.baz·&&·foo.bar?.qux;
+      │         -----------                      
```

Comment:
I've been wondering this since my previous attempt to fix this rule: Why sometimes the suggested fix doesn't provide a diff (like what it does in the following diagnostics)?

Reply:
Oh yes, that tripped me in the beginning too. The trick is, when it *only* removes characters it is indicated by minuses below. Once you know it is a bit clearer, but it is not intuitive at first, I agree.

Reply:
I know, I had the exact same experience 😅